plugins/formal-verification/formver.viper/src/org/jetbrains/kotlin/formver/viper/errors/VerificationError.kt

```diff
@@ -10,8 +10,17 @@ import org.jetbrains.kotlin.formver.viper.ast.Position
 import org.jetbrains.kotlin.formver.viper.ast.info
 import org.jetbrains.kotlin.formver.viper.ast.unwrapOr
 
+/**
+ * This class acts as wrapper for Viper's [viper.silver.verifier.ErrorReason].
+ * This is necessary since extension functions on [viper.silver.verifier.ErrorReason] cannot be
+ * used outside the class' package.
+ */
+data class ErrorReason(val reason: viper.silver.verifier.ErrorReason)
+
 interface VerificationError : VerifierError {
```

Comment:
non-blocking: One thing I'm not clear on: is this interface still adding any value?  It seems like a class could do just as well.

Alternatively, why have `ErrorAdapter` construct an object like that and not just make one specific implementation that takes a `VerificationResult`?

Reply:
Good point! I was thinking to do do something like this:
```kotlin
class VerificationError private constructor(
    val result: viper.silver.verifier.VerificationError
) : VerifierError {
    companion object {
        fun fromSilver(result: viper.silicon.interfaces.VerificationResult): VerificationError {
            check(result.isFatal) { "The verification result must contain an error to be converted." }
            return VerificationError((result as viper.silicon.interfaces.Failure).message())
        }
    }
}
```